### DISCUSSION TITLE
Option for gateway to store funds in account, not ecash

### DISCUSSION THREAD

**Discussion by justinmoon** [fedimint/fedimint#1715] [General]
Gateways get less benefit from ecash than end-user clients because gateway pubkeys are visible in federation contracts anyway.

If gateways could store their funds in an account and not ecash, then we could save a round of consensus for every successful lightning payment (blind signature shares) and also grow the federation spend book more slowly.

**Comment by elsirion** [comment]
Under normal circumstances the velocity of money in the gateway shouldn't be a limiting factor. The gateway can already return success to the user before getting its money (https://github.com/fedimint/fedimint/issues/1595#issuecomment-1419770358). Furthermore, the gateway using e-cash notes still adds to the anonymity set.

There actually already is a plain account type in the LN module btw, it's just not tested at all and should probably be removed.

https://github.com/fedimint/fedimint/blob/4380bfe02447673ab89f14b23a3c6a256591caee/modules/fedimint-ln/src/contracts/account.rs#L7-L11
